File: cb/dtos/credential.py
from __future__ import annotations
"""Credential DTOs -- UsernamePassword focused."""

import dataclasses
from cb.dtos.base import BaseDTO


@dataclasses.dataclass
class CredentialDTO(BaseDTO):
    id: str = ""
    display_name: str = ""
    type_name: str = ""
    scope: str = "GLOBAL"
    description: str = ""

    @classmethod
    def from_dict(cls, data: dict) -> "CredentialDTO":
        return cls(
            id=data.get("id", ""),
            display_name=data.get("displayName", ""),
            type_name=data.get("typeName", ""),
            scope=data.get("scope", "GLOBAL"),
            description=data.get("description", ""),
        )

File: src/core/dtos/credential.ts
/**
 * Credential DTOs.
 * Mirrors legacy/cb/dtos/credential.py
 */

import { str } from "./base.js";

/** Shape returned by a credential store's `credentials[]` entries. */
export interface CredentialDTO {
  id: string;
  displayName: string;
  typeName: string;
  scope: string;
  description: string;
}

/** Maps a raw credential API entry to CredentialDTO; `scope` defaults to `"GLOBAL"` when blank. */
export function credentialFromDict(data: Record<string, unknown>): CredentialDTO {
  return {
    id: str(data["id"]),
    displayName: str(data["displayName"]),
    typeName: str(data["typeName"]),
    scope: str(data["scope"]) || "GLOBAL",
    description: str(data["description"]),
  };
}
